tests/lib/rules/no-constant-condition.js

```diff
@@ -90,6 +89,19 @@ ruleTester.run("no-constant-condition", rule, {
         "if ((foo || 233) <= 666) {}",
         "if ((key || 'k') in obj) {}",
         "if ((foo || {}) instanceof obj) {}",
+        "if ((foo || 'bar' || 'bar') === 'bar');",
+        {
+            code: "if ((foo || 1n) === 'baz') {}",
+            parserOptions: { ecmaVersion: 11 }
+        },
+        {
+            code: "if (a && 0n || b);",
+            parserOptions: { ecmaVersion: 11 }
+        },
```

Comment:
it seems weird to not report this case. it can be written to `if(b);` ?

Reply:
It's okay now because the whole `a && 0n || b` isn't constant.

This would be reported by one of the enhancements proposed in https://github.com/eslint/eslint/issues/13752. If the rule checks the left operand of `||` as a condition for itself, then it would find out that it's always falsy.